lib/bandit/pipeline.ex

```diff
@@ -88,6 +88,16 @@ defmodule Bandit.Pipeline do
        ),
        do: {:ok, to_string(host), port || local_port}
 
+  @spec determine_local_port(ThousandIsland.Transport.socket_info()) :: integer()
+  defp determine_local_port(local_info) do
+    case local_info do
+      {:local, _} -> 0
+      {:unspec, _} -> 0
+      {:undefined, _} -> 0
```

Comment:
`{type, _} when type in [:local, :unspec, :undefined] -> 0` would avoid the repetition of the default port.